src/main/java/com/hsbc/engineering/Arguments.java

```diff
@@ -91,15 +89,15 @@ public Arguments(Map<String, String> systemEnvs) {
      */
     public Map<String, String> set(Map<String, String> systemEnvs) {
         final boolean bool_run_extended_tests = Boolean.parseBoolean(systemEnvs.get(RUN_EXTENDED_TESTS));
-        String sqlQuery = isNullAndEmpty(systemEnvs.get(SQL)) ? DEFAULT_SQL : systemEnvs.get(SQL);
+        String sqlQuery = isNullOrEmpty(systemEnvs.get(SQL)) ? DEFAULT_SQL : systemEnvs.get(SQL);
 
         arguments = new HashMap<>() {{
                 put(CLASS_NAME, systemEnvs.get(CLASS_NAME));
                 put(URL, systemEnvs.get(URL));
                 put(RUN_EXTENDED_TESTS, Boolean.toString(bool_run_extended_tests));
                 put(RUN_PERFORMANCE_EXTRACTION_TEST, Boolean.toString(bool_run_performance_extraction_test));
                 put(SQL, DEFAULT_SQL);
-            }};
+        }};
```

Comment:
arguments = new HashMap<>(Map.of(
                CLASS_NAME, systemEnvs.get(CLASS_NAME),
                URL, systemEnvs.get(URL),
                RUN_EXTENDED_TESTS, Boolean.toString(bool_run_extended_tests),
                RUN_PERFORMANCE_EXTRACTION_TEST, Boolean.toString(bool_run_performance_extraction_test),
                SQL, DEFAULT_SQL
        )); // Because Map.of() is immutable, so we need to wrap it with a new HashMap()